src/core/geometry/qgsgeometry.cpp

```diff
@@ -829,6 +829,15 @@ QgsGeometry::OperationResult QgsGeometry::splitGeometry( const QgsPointSequence
 
   QVector<QgsGeometry > newGeoms;
   QgsLineString splitLineString( splitLine );
+
+  /**
+   * QGIS uses GEOS algorithm to split geometries.
+   * Using 3D points in GEOS will returns an interpolation value which is the
+   * mean between geometries.
+   * On the contrary, in our logic, the interpolation is a linear interpolation
+   * on the split point. By dropping Z/M value, GEOS will returns the expected
+   * result. See https://github.com/qgis/QGIS/issues/33489
+   */
```

Comment:
Can you put the docstring to the header so it's visible in the api docs?
It shouldn't mention any implementation details like geos and mainly focus on the expected result. The one here is very nice as reference for future readers!